### PR TITLE
Fixed transpose issue

### PR DESCRIPTION
**Pull request by irischang020**

### CODE REVIEW

**Comment by irischang020** [comment]
diffraction pattern of GroEL-GroES generated using skopi at a predetermined orientation (total photons on detector = 533375.0)
image

diffraction pattern of GroEL-GroES generated using condor at the same orientation (total photons on detector = 543532)
image

File: skopi/detector/simple_square.py [hunk 1]
```diff
@@ -28,7 +28,7 @@ def __init__(self, N_pixel, det_size, det_distance, beam=None):
         sep = float(det_size) / N_pixel
         x = -det_size/2 + sep/2 + ar*sep
         y = -det_size/2 + sep/2 + ar*sep
-        X, Y = xp.meshgrid(x, y, indexing='xy')
+        Y, X = xp.meshgrid(x, y, indexing='xy')
```
Comment (chuckie82) [inline skopi/detector/simple_square.py:31]: x and y are exactly the same here
therefore X and Y must be the same
so how does changing X,Y to Y,X help?

Comment (irischang020) [reply skopi/detector/simple_square.py:31]: I looked into condor and they swapped X and Y.
```
def generate_xypix(self, cx=None, cy=None):
        Y, X = numpy.meshgrid(numpy.float64(numpy.arange(self._ny))-(0. if cy is None else cy),
                              numpy.float64(numpy.arange(self._nx))-(0. if cx is None else cx),
                              indexing="ij")
        return X, Y
```

File: skopi/detector/simple_square.py [hunk 2]
```diff
@@ -28,7 +28,7 @@ def __init__(self, N_pixel, det_size, det_distance, beam=None):
...
         Xar, Yar = xp.meshgrid(ar, ar, indexing='xy')
```
Comment (chuckie82) [inline skopi/detector/simple_square.py:32]: Feels inconsistent to have Y,X in line 31 and Xar,Yar in line 32.
I suppose changing to Yar,Xar will give wrong results?

Comment (irischang020) [reply skopi/detector/simple_square.py:32]: Changing Xar,Yar to Yar,Xar, the result becomes the following (the same as keeping the original):
image

File: skopi/detector/simple_square.py (hunk 1 above)
Comment (chuckie82) [reply skopi/detector/simple_square.py:31]: they are using indexing="ij" and we are using indexing="xy" which is equivalent to a transpose

>>> x=range(3)
>>> y=range(3)
>>> xv, yv = np.meshgrid(x, y, sparse=False, indexing='ij')
>>> xv
array([[0, 0, 0],
       [1, 1, 1],
       [2, 2, 2]])
>>> yv
array([[0, 1, 2],
       [0, 1, 2],
       [0, 1, 2]])
>>> xv, yv = np.meshgrid(x, y, sparse=False, indexing='xy')
>>> xv
array([[0, 1, 2],
       [0, 1, 2],
       [0, 1, 2]])
>>> yv
array([[0, 0, 0],
       [1, 1, 1],
       [2, 2, 2]])

Comment (irischang020) [reply skopi/detector/simple_square.py:31]: changed to
```
        X, Y = xp.meshgrid(x, y, indexing='ij')
        Xar, Yar = xp.meshgrid(ar, ar, indexing='ij')
```
image

**Comment by apeck12** [comment]
@irischang020, I think this proposed change to the SimpleSquareDetector will make it inconsistent with the other detectors. Here's are the images I'm getting for a few different detectors with the code before this commit:

download-3

**Comment by chuckie82** [comment]
@irischang020 was @apeck12 's comment above addressed?

**Comment by irischang020** [comment]
If we want skopi to be internally consistent, then we can just keep the current implementation of the SimpleSquareDetector.

**Pull request closed without merging**